### DISCUSSION TITLE
Python爬蟲- 使用header假裝自己是瀏覽器

### DISCUSSION THREAD

**Discussion by gnsJhenJie** [1101-GS4719-C/Python-Community-GS4719-C#615] [General]
剛剛在上課的時候發現照著老師的程式碼執行會被Dcard擋掉，但實際上使用瀏覽器造訪[dcard.tw](https://dcard.tw)時卻不會被擋，明明是同樣的IP造訪Dcard，卻有不同的結果，可見Dcard的封鎖機制不單單只透過IP來封鎖，再搜尋後發現可以透過Python request的header欄位來假裝自己是瀏覽器，發送與瀏覽器更相近的request，可以參考下面的code:
```python
import requests as rs
url = 'https://www.dcard.tw/service/api/v2/forums/movie/posts?limit=30&before=238897021'
headers = { 'content-type': 'text/html; charset=UTF-8', 'user-agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/76.0.3809.132 Safari/537.36'
}
posts = rs.get(url,headers=headers)
# print(posts.text)
jsonobj = posts.json()
print(jsonobj[0].items())
```

可以發現有用header(上)和沒用header(下)的差別:
<img width="1497" alt="Screen Shot 2022-05-18 at 15 44 25" src="https://user-images.githubusercontent.com/32983347/168986419-ed067bab-e2ed-41af-8b46-15848bf85876.png">

**Comment by sys511613** [comment]
最後老師有講可以加上header這個方法了 不過還是感謝分享
